src/components/base/outline-button/outline-button.ts

```diff
@@ -49,19 +55,105 @@ export class OutlineButton extends OutlineElement {
    * Whether the button is disabled. Only applies to
    * implementations not using the url property
    */
-  @property({ type: Boolean }) disabled: boolean;
+  @property({ type: Boolean }) isDisabled = false;
+
+  /**
+   * A click handler to be passed only to onClick. DO NOT USE @click on this component.
+   */
+  @property() onClick: () => void;
+
+  /**
+   * A keyUp handler to be passed to the onKeyUp. DO NOT USE @keyup on this component.
+   */
+  @property() onKeyUp: () => void;
+
+  /**
+   * Sets tabindex attribute to default "-1", for accessibility purposes.
+   */
+  @property({ type: String, reflect: true }) tabindex = '-1';
+
+  /**
+   * Sets the text for screen readers and tooltips for disabled buttons.
+   * If your button can be disabled you must provide a short explanitory reason.
+   */
+  @property({ type: String }) disabledInfo?: string | undefined;
+
+  /**
+   * Sets the text for screen readers and tooltips. .
+   */
+  @property({ type: String }) tooltipInfo?: string | undefined;
+
+  /**
+   * Sets the position of the tooltipInfo/disabledInfo display on hover.
+   */
+  @property({ type: String }) position: TooltipPosition = 'bottom';
+
+  /**
+   * Used to provied a unique ID to panel content <div> for accessibility purposes.
+   */
+  seed = Math.floor(Math.random() * 10000);
```

Comment:
Could this be part of the logic in the `outline-tooltip` component? Are there cases when it's helpful for the consumer to provide the ID or could be generate random strings in all situations?

Reply:
tooltip requires an id so that the target can be associated with the tooltip. It seemed best for the id to be user/consumer generated incase they wanted/needed to specify. In this scenario hardcode the behavior. But I'm open to removing the id prop entirely from the tooltip and letting it be autogenerated within the tooltip itself.